capstone/src/main/java/com/google/sps/servlets/StudentServlet.java

```diff
@@ -26,4 +26,29 @@ private static String convertToJsonUsingGson(Student student) {
     String json = gson.toJson(student);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get student object based on the logged in email
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    Student student = PrototypeStudents.PROTOTYPE_STUDENTS.get(userEmail);
+
+    if (request.getParameter("action").equals("join")) {
+      String clubToJoin = request.getParameter("club");
+      if (clubToJoin != null && !clubToJoin.isEmpty()) {
+        student.joinClub(clubToJoin);
```

Comment:
I'm not quite sure what the purpose of this change is. Could you elaborate on this more (also a good example of how a more detailed PR description like Linh mentioned is helpful)? Is it to add/remove clubs permanently to the student object? If so, I'm not sure this is accomplishing that, since "student" is a local variable to this request and "disappears" after the method returns. 

Also, I might have missed it but I can't seem the implementation for student.joinClub and student.removeClub. Are these methods defined somewhere?

Reply:
Yeah, the purpose of this code is to permanently change the student object. I know Megha was working on code to add/remove a club from the student, so I'm realizing this is a premature PR - I should have waited for Megha's code to merge to finish this bit and push.

Reply:
Yeah, looking at Megha's PR now and I just realized that. Will add some comments on her PR, but some of those comments will apply here after I finish that review.

Reply:
Sounds good - I'll make sure to look at the comments on Megha's PR before I go about fixing/completing mine.